service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/MessageJudgeController.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.NewJudicialMessageEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.JudicialMessage;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.MessageJudgeService;
+import uk.gov.hmcts.reform.fpl.utils.CaseDetailsMap;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.removeTemporaryFields;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsMap.caseDetailsMap;
+
+@Api
+@RestController
+@RequestMapping("/callback/message-judge")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class MessageJudgeController extends CallbackController {
+    private final MessageJudgeService messageJudgeService;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+        CaseDetailsMap caseDetailsMap = caseDetailsMap(caseDetails);
+
+        caseDetailsMap.putAll(messageJudgeService.initialiseCaseFields(caseData));
+        removeTemporaryFields(caseDetailsMap, "judicialMessageMetaData", "judicialMessageNote");
```

Comment:
I think that these could be removed with the others in the about-to-submit. You seem to need it for the notification but you can get the latest information from either relying on the ordering of the message list (most recent updated will be at the top of the list) or get the difference of the before and current list to find the element that has changed. This can then be passed to the event and all transient properties can be cleaned up in the about-to-submit